frontend/test/metabase/scenarios/visualizations/reproductions/11727-cancel-native-query-shortcut.cy.spec.js

```diff
@@ -0,0 +1,38 @@
+import {
+  restore,
+  withDatabase,
+  adhocQuestionHash,
+} from "__support__/e2e/cypress";
+
+const PG_DB_ID = 2;
+
+const questionDetails = {
+  dataset_query: {
+    type: "native",
+    database: PG_DB_ID,
+    native: {
+      query: "SELECT pg_sleep(10)",
+    },
+  },
+};
+
+describe("issue 11727", () => {
+  beforeEach(() => {
+    restore("postgres-12");
+    cy.signInAsAdmin();
+    cy.intercept("GET", "/api/database").as("getDatabases");
+  });
+
+  it("should cancel the native query via the keyboard shortcut (metabase#11727)", () => {
+    withDatabase(PG_DB_ID, () => {
+      cy.visit(`/question#` + adhocQuestionHash(questionDetails));
```

Comment:
We cannot use `visitQuestionAdhoc` here because it waits for the `dataset` call, but we need to cancel the query before it is executed.